guava/src/com/google/common/escape/UnicodeEscaper.java

```diff
@@ -242,7 +246,7 @@ protected final String escapeSlow(String s, int index) {
    * @return the Unicode code point for the given index or the negated value of the trailing high
    *     surrogate character at the end of the sequence
    */
-  protected static int codePointAt(CharSequence seq, int index, int end) {
+  protected static int codePointAt(CharSequence seq, @IndexOrHigh("#1") int index, @IndexFor("#1") int end) {
```

Comment:
@NhatDinh, why did you change the annotation of `index` to `@IndexOrHigh`, contrary to my suggestion?

Reply:
@panacekcz it was an accident.

Reply:
@panacekcz after taking another look, I think `int end` should be annotated as `@IndexFor("#1")` as before. Because for example if seq.length = 5, index = 4, end = 5.  Then line 252 `char c1 = seq.charAt(index++);`  will be executed and cause an error( index++ is 5 and seq.charAt(5) will cause error).